iroh-cli/src/commands/gossip.rs

```diff
@@ -0,0 +1,71 @@
+use anyhow::{Context, Result};
+use bao_tree::blake3;
+use clap::Subcommand;
+use futures_lite::StreamExt;
+use futures_util::SinkExt;
+use iroh::client::{Iroh, RpcService};
+use iroh::net::NodeId;
+use quic_rpc::ServiceConnection;
+use tokio::io::AsyncBufReadExt;
+
+#[derive(Subcommand, Debug, Clone)]
+#[allow(clippy::large_enum_variant)]
+pub enum GossipCommands {
+    /// Subscribe to a topic
+    Subscribe {
+        #[clap(long)]
+        topic: String,
+        bootstrap: Vec<NodeId>,
+        #[clap(long, short)]
+        verbose: bool,
+    },
+}
+
+impl GossipCommands {
+    pub async fn run<C>(self, iroh: &Iroh<C>) -> Result<()>
+    where
+        C: ServiceConnection<RpcService>,
+    {
+        match self {
+            Self::Subscribe {
+                topic,
+                bootstrap,
+                verbose,
+            } => {
+                let bootstrap = bootstrap.into_iter().collect();
+                let topic = blake3::hash(topic.as_ref()).into();
```

Comment:
Not sure. Maybe we should just allow people to specify the exact gossip topic instead of hiding it. I was just too lazy to add a new command line arg.

Change the RPC to contain an actual topic-id, and have alternative CLI options topic: String (gets hashed on the client side) and --topic_raw: TopicId ?